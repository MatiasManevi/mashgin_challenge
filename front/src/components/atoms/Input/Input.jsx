import React from 'react';

import './Input.scss';

const Input = ({ changeHandler, value, placeholder = '' }) => {
	return (
		<input
			onChange={changeHandler}
			className="input"
			type="text"
			value={value}
			placeholder={placeholder}
		/>
	);
};

export default Input;
