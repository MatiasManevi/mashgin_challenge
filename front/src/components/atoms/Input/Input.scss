@import 'assets/scss/variables';

.input {
	width: 100%;
	font-size: $font-size-24;
	padding-left: $space-16;
	padding-bottom: $space-4;
	padding-top: $space-4;
	height: $space-32;
	outline: none;
	border: none;
}
