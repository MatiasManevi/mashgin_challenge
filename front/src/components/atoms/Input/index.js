export { default } from 'components/atoms/Input/Input';
