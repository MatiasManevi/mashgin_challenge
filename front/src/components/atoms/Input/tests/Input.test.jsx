import React from 'react';
import { shallow } from 'enzyme';

import Input from '../Input';

describe('Input', () => {
	const changeHandler = jest.fn();
	const value = 'input value';

	it('should match snapshot', () => {
		const wrapper = shallow(
			<Input changeHandler={changeHandler} value={value} />
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('should call changeHandler when onChange', () => {
		const event = {
			target: { value }
		};
		const wrapper = shallow(
			<Input changeHandler={changeHandler} value={value} />
		);
		wrapper.simulate('change', event);
		expect(changeHandler).toHaveBeenCalledTimes(1);
		expect(changeHandler).toHaveBeenCalledWith(event);
	});
});
