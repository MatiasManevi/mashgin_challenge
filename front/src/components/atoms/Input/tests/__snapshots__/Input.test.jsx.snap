// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<input
  className="input"
  onChange={[MockFunction]}
  placeholder=""
  type="text"
  value="input value"
/>
`;
